android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt

```diff
@@ -58,15 +51,28 @@ class HomePageViewModel @Inject constructor(
 		FilterVals(
 			locationRange = locationRange,
 			priceRange = priceRange,
-			roomRange = roomRange
+			roomRange = roomRange,
 		)
 	}
+
+	private val listingsStream: Observable<GetListingsResponse> = filterStream.map {
+		runBlocking {
+			api.listingsList(
+				priceMin = null,
+				priceMax = null,
+				roomsMin = null,
+				roomsMax = null,
+			)
+		}
+	}
+		.subscribeOn(Schedulers.io())
```

Comment:
I am totally not sure, but is there any error handling for this API call?

Reply:
Nope, I totally forgot to add it 😅 
Good catch, would have been pretty bad if we let this through